app/src/main/java/me/phh/treble/app/Lid.kt

```diff
@@ -26,6 +27,38 @@ object Lid: EntryStartup {
         PowerManager::class.java.getMethod("goToSleep", Long::class.java).invoke(powerManager, SystemClock.uptimeMillis())
     }
 
+
+    fun huawei(ctxt: Context) {
+        val sensorManager = ctxt.getSystemService(SensorManager::class.java)
+        // Huawei Hall sensor for magnetic cover : type = 65538 - name = "HALL sensor"
+        // find the first item or return null
+        // type=65538 is the Huawei type for magnetic sensor (found in the EMUI9 frameworks)
+        val lidSensor = sensorManager.getSensorList(Sensor.TYPE_ALL).find { it.type == 65538 }
```

Comment:
I meant to use `stringType`, which will be more specific, not `type` which might collide on other devices

Reply:
Ok sorry, but there can't be any conflict, as I marked in comment, this type is exclusively reserved for cover. This is exactly how huawei does in its emui framework, it is a reserved type

Reply:
And then we'll get people who fake huawei fingerprint (because somehow at that time that's the way to pass PITA, or because they need it for some app or just because they do weird things like users do) and who will complain of screen randomly powering off, while we could just use the stringType.

I can write that code if you prefer, just need to grab that stringtype from dumpsys sensorservice

Reply:
Yes I would like to because I don't understand, there must be something that is beyond me, but you are the expert, give me the code and I will test it.

Reply:
I will get dumpsys

Reply:
done

Reply:
try with
`val lidSensor = sensorManager.getSensorList(Sensor.TYPE_ALL).find { it.stringType == "android.sensor.hall" }`